pkg/apis/experimental/v1/types.go

```diff
@@ -462,3 +462,79 @@ type JobCondition struct {
 	// Human readable message indicating details about last transition.
 	Message string `json:"message,omitempty"`
 }
+
+// Ingress encapsulates the inputs needed to config the load-balancer.
+// It represents a rule collection of inbound connections from the external network
+// that would be satisfied by a load-balancer.
+type Ingress struct {
+	v1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata
+	v1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the desired state of the Ingress.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#spec-and-status
+	Spec IngressSpec `json:"spec,omitempty"`
+
+	// Status is the current state of the Ingress.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#spec-and-status
+	Status IngressStatus `json:"status,omitempty"`
+}
+
+// IngressList is a collection of Ingress.
+type IngressList struct {
+	v1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata
+	v1.ListMeta `json:"metadata,omitempty"`
+
+	// Items is the list of ingress.
+	Items []Ingress `json:"items"`
+}
+
+// IngressSpec describes the ingress the user wishes to exist.
+type IngressSpec struct {
+	// PathMap defines a list of rules which mapping multiple hosts and paths to the backend services.
+	// The rules will be used to configure the corresponding load-balancer.
+	// Example: Host + Path -> Service Backend (Service Name + Service Protocol + Service Port)
+	PathMap []IngressMap `json:"pathMap,omitempty"`
```

Comment:
I'd be fine to move it around some, though I am a bit confused on the current state of things - the PR doesn't reflect the most recent comments I have seen, so I have lost track of what it "currently" looks like.

Also, this is in experimental, so a less-than-perfect something for v1.1 is better than a perfect nothing.  we can iterate